src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutUnpatchedVulnerabilities.java

```diff
@@ -0,0 +1,65 @@
+package com.sap.sgs.phosphor.fosstars.data.interactive;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.VULNERABILITIES;
+import static com.sap.sgs.phosphor.fosstars.model.value.Vulnerability.NO_DESCRIPTION;
+import static com.sap.sgs.phosphor.fosstars.model.value.Vulnerability.NO_REFERENCES;
+import static com.sap.sgs.phosphor.fosstars.model.value.Vulnerability.UNKNOWN_FIXED_DATE;
+import static com.sap.sgs.phosphor.fosstars.model.value.Vulnerability.UNKNOWN_INTRODUCED_DATE;
+import static com.sap.sgs.phosphor.fosstars.tool.YesNoQuestion.Answer.YES;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.CVSS;
+import com.sap.sgs.phosphor.fosstars.model.value.Vulnerabilities;
+import com.sap.sgs.phosphor.fosstars.model.value.Vulnerability;
+import com.sap.sgs.phosphor.fosstars.model.value.Vulnerability.Resolution;
+import com.sap.sgs.phosphor.fosstars.tool.InputURL;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoQuestion;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoQuestion.Answer;
+
+/**
+ * This data provider asks a user about unpatched vulnerabilities.
+ */
+public class AskAboutUnpatchedVulnerabilities<T> extends AbstractInteractiveDataProvider<T> {
+
+  @Override
+  protected AskAboutUnpatchedVulnerabilities<T> ask(T object, ValueSet values) {
+    Vulnerabilities unpatchedVulnerabilities = new Vulnerabilities();
+
+    Answer answer = new YesNoQuestion(
+        callback, "Are you aware about any unpatched vulnerability in the project?").ask();
+
+    if (answer == YES) {
+      do {
+        callback.say("[+] This is awesome! Or, no?..");
+        callback.say("[?] Please give me a URL for the vulnerability");
+        String id = new InputURL(callback).get().toString();
+        Vulnerability vulnerability = new Vulnerability(
+            id, NO_DESCRIPTION, CVSS.UNKNOWN, NO_REFERENCES, Resolution.UNPATCHED,
+            UNKNOWN_INTRODUCED_DATE, UNKNOWN_FIXED_DATE);
+        unpatchedVulnerabilities.add(vulnerability);
+        answer = new YesNoQuestion(callback, "One more?").ask();
+      } while (answer == YES);
+    }
+
+    Value<Vulnerabilities> vulnerabilities = knownVulnerabilities(values);
+    vulnerabilities.get().add(unpatchedVulnerabilities);
+    values.update(vulnerabilities);
+
+    // TODO: store the entered vulnerabilities somewhere
```

Comment:
Is there an issue tracking this?

Reply:
No, we don't have an issue for that. To be honest, I am not sure if it should be implemented. The interactive providers are not a priority for now.